lib/web/integrations_awsoidc.go

```diff
@@ -1255,3 +1221,130 @@ func (h *Handler) awsOIDCListSubnets(w http.ResponseWriter, r *http.Request, p h
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	serviceURLByVPC, err := getServiceURLs(dbServices, req.AccountID, req.Region, h.auth.clusterName)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	vpcs := make([]ui.DatabaseEnrollmentVPC, 0, len(listResp.Vpcs))
+	for _, vpc := range listResp.Vpcs {
+		vpcs = append(vpcs, ui.DatabaseEnrollmentVPC{
+			VPC: awsoidc.VPC{
+				Name: vpc.Name,
+				ID:   vpc.Id,
+			},
+			ECSServiceDashboardURL: serviceURLByVPC[vpc.Id],
+		})
+	}
+
+	return ui.AWSOIDCDatabaseVPCsResponse{
+		NextToken: listResp.NextToken,
+		VPCs:      vpcs,
+	}, nil
+}
+
+func fetchAWSOIDCDatabaseServices(ctx context.Context, clt client.GetResourcesClient) ([]types.DatabaseService, error) {
+	// Get all database services with the AWS OIDC agent metadata label.
+	var nextToken string
+	var fetchedDbSvcs []types.DatabaseService
+	for {
+		page, err := client.GetResourcePage[types.DatabaseService](ctx, clt, &proto.ListResourcesRequest{
+			ResourceType: types.KindDatabaseService,
+			Limit:        defaults.MaxIterationLimit,
+			StartKey:     nextToken,
+			Labels:       map[string]string{types.AWSOIDCAgentLabel: types.True},
+		})
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		fetchedDbSvcs = append(fetchedDbSvcs, page.Resources...)
+		nextToken = page.NextKey
+		if len(nextToken) == 0 {
+			return fetchedDbSvcs, nil
+		}
+	}
+}
+
+// getDBServiceVPC returns the database service's VPC ID selector value if the
+// database service was deployed by the AWS OIDC integration, otherwise it
+// returns an empty string.
+func getDBServiceVPC(svc types.DatabaseService, accountID, region string) string {
+	if len(svc.GetResourceMatchers()) != 1 || svc.GetResourceMatchers()[0].Labels == nil {
+		return ""
+	}
+
+	// Database services deployed by Teleport have known configurations where
+	// we will only define a single resource matcher.
+	labelMatcher := *svc.GetResourceMatchers()[0].Labels
+
+	// We check for length 3, because we are only
+	// wanting/checking for 3 discovery labels.
+	if len(labelMatcher) != 3 {
+		return ""
+	}
+	if slices.Compare(labelMatcher[types.DiscoveryLabelAccountID], []string{accountID}) != 0 {
+		return ""
+	}
+	if slices.Compare(labelMatcher[types.DiscoveryLabelRegion], []string{region}) != 0 {
+		return ""
+	}
+	if len(labelMatcher[types.DiscoveryLabelVPCID]) != 1 {
+		return ""
+	}
+	return labelMatcher[types.DiscoveryLabelVPCID][0]
+}
+
+// getServiceURLs returns a map vpcID -> service URL for ECS services deployed
+// by the OIDC integration in the given account and region.
+func getServiceURLs(dbServices []types.DatabaseService, accountID, region, teleportClusterName string) (map[string]string, error) {
+	serviceURLByVPC := make(map[string]string)
+	for _, svc := range dbServices {
+		vpcID := getDBServiceVPC(svc, accountID, region)
+		if vpcID == "" {
+			continue
+		}
+		svcURL, err := awsoidc.ECSDatabaseServiceDashboardURL(region, teleportClusterName, vpcID)
```

Comment:
would it be more helpful to user, if we take this further and add the specific vpc into the URL? so it's not just a URL to the dashboard with a list of vpcs?

before, we couldn't provide a URL to a specific vpc, since there could be multiple deployments, speaking of which, we might want to change this too when we `deploy service`?

Reply:
I'm not sure I understand - we deploy an ECS service for each VPC and include the VPC id in the service's name, so this is a link to the ECS service for a specific VPC

Reply:
sorry disregard, i somehow completely missed that you are adding the vpc id to the URL 😅